sql/plan/block.go

```diff
@@ -86,65 +85,117 @@ func (b *Block) WithChildren(children ...sql.Node) (sql.Node, error) {
 
 // RowIter implements the sql.Node interface.
 func (b *Block) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	var returnRows []sql.Row
+	var returnNode sql.Node
+	var returnSch sql.Schema
+
+	selectSeen := false
+	for _, s := range b.statements {
```

Comment:
It would be better if you could figure out how to do this work one row at a time in a real iter, instead of all up front. But this is fine for a first pass